sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -297,6 +298,22 @@ public void setDecimal(int ordinal, Decimal value, int precision) {
     }
   }
 
+  @Override
+  public void setInterval(int ordinal, CalendarInterval value) {
+    assertIndexIsValid(ordinal);
+    long cursor = getLong(ordinal) >>> 32;
+    assert cursor > 0 : "invalid cursor " + cursor;
+    if (value == null) {
+      setNullAt(ordinal);
+      Platform.putLong(baseObject, getFieldOffset(ordinal), (cursor << 32) | 16L);
```

Comment:
in `setDecimal` we use `cursor << 32`. Why change it?

Reply:
I have no strong opinion here, I do this just to make a reservation for the interval values in the variable-length portion.  I guess this won't be needed

Reply:
This is different from decimal, which is var-length. We know the length is always 16 for interval values.

Anyway, it doesn't hurt to encode the length here. Let's keep it.